### DISCUSSION TITLE
gatsby-source-shopify: What's the difference between product `id` and `shopifyId` in GraphQL?

### DISCUSSION THREAD

**Discussion by dungle-scrubs** [gatsbyjs/gatsby#30706] [Help]
Hi,

When querying a product, what is the purpose of `id` as opposed to `shopifyId`?

I see when I use `shopify-gid` to decode `shopifyId`, I get a product object with the original product ID, but a little confused about `id` in the query (which looks something like `Shopify__Product__Z2lkOi8vc2hvcGlmeS9Qdm9kdWN0LzQ3MTY3Mzc0MDY2MDA=`)

Thanks

**Comment by LekoArts** [comment]
`id` is a reserved field in our Gatsby node store, `shopifyId` is the `id` coming from Shopify's API. As `id` is reserved it's converted to `shopifyId`

**Reply by dungle-scrubs** [reply]
Thanks for clarifying
